server/src/main/java/org/opensearch/identity/noop/NoopTokenManager.java

```diff
@@ -26,8 +28,17 @@ public class NoopTokenManager implements TokenManager {
      * @return a new Noop Token
      */
     @Override
-    public AuthToken issueToken(String audience) {
+    public AuthToken issueOnBehalfOfToken(final Subject subject, final OnBehalfOfClaims claims) {
         return new AuthToken() {
+            @Override
+            public String getTokenValue() {
+                return "noopToken";
+            }
         };
     }
+
+    @Override
+    public Subject authenticateToken(AuthToken authToken) {
+        return null;
+    }
```

Comment:
I don't think the TokenManager should know how to find subjects in other IDPs, this should be removed in place of Subject's authenticate method used in its place, no?

Reply:
I think a token manager should be able to validate a token it has issued. It is the source of truth for the token and will need to be able to wrap whatever authentication the token is ultimately a part of. 

If we issue an OnBehalfOf token with the OnBehalfOf Token Manager, we then need to be able to handle that token's authentication with the token manager. Though OBO tokens are simply JWTs, there is nothing stopping us from making a token manager that hashes another auth system and then hands out tokens as strings. It would then need to be able to look at those strings and resolve them for the Identity Plugin. 

If we made this pointless but valid token manager: 

```
weirdTokenManager {

String salt = "SuperSecretSalt"

public authToken issueToken(string info): 
   
   BearerToken baseToken = new BearerToken(info)
   
   return new saltedAuthToken(baseToken, salt;
 }
 
 ```
 
 We would need to be able to authenticate the token. This would be impossible without having the token manager authenticate the token since it is the only thing that knows the salt.